SmartTutor/ContentModel/LearningObjects/Challenge.cs

```diff
@@ -0,0 +1,47 @@
+﻿using RepositoryCompiler.CodeModel.CaDETModel.CodeItems;
+using SmartTutor.ContentModel.LearningObjects.MetricRules;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SmartTutor.ContentModel.LearningObjects
+{
+    public class Challenge : LearningObject
+    {
+        public string Url { get; internal set; }
+        public List<CaDETClass> ResolvedClasses { get; internal set; }
+        public List<MetricRangeRule> ClassMetricRules { get; internal set; }
+        public List<MetricRangeRule> MethodMetricRules { get; internal set; }
+
+        // TODO: This is basic strategy, implement strategy pattern interface 
+        public bool CheckSubmittedChallengeCompletion(List<CaDETClass> submittetClasses)
+        {
+            List<CaDETMember> submittedMethods = GetMethodsFromClasses(submittetClasses);
+            if (!CheckClassesCompletion(submittetClasses) || !CheckMethodsCompletion(submittedMethods))
```

Comment:
Lines 19-21 can be transformed into "return CheckClassesCompletion(submittetClasses) && CheckMethodsCompletion(submittedMethods)".

Always pay attention when you are returning booleans based on a branch - usually you can return some variation of the conditional for the branch.